src/assets/theme/ThemeProvider.tsx

```diff
@@ -0,0 +1,25 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import React, { ReactElement, ReactChild, ReactChildren } from "react";
+import { ThemeProvider } from "styled-components";
+import { connect } from "react-redux";
+
+import { ApplicationStateModel } from "../../state/application-state.model";
+import { ThemeKey, themes } from ".";
+
+interface TPR {
+  themeKey: ThemeKey;
+  children: ReactChild | ReactChildren | ReactElement;
+}
+
+const TProvider: React.FC<TPR> = ({ themeKey, children }: TPR) => (
+  <ThemeProvider theme={themes[themeKey]}>{children}</ThemeProvider>
+);
+
+const mapSTP = (store: ApplicationStateModel) => ({
+  themeKey: store.theme.key,
+});
+
+export default connect(mapSTP)(TProvider);
```

Comment:
Trochę nie rozumiem dlaczego używasz `connect`.

`mapSTP` to funkcyjny komponent, więc możemy użyć w środku `useSelector` i preferowałbym takie rozwiązanie, ponieważ takiego API używam w innych miejscach aplikacji, natomiast `connect()` może być nieznany dla niektórych osób.

Chyba że czegoś nie rozumiem, i tutaj po prostu nie da się użyć `useSelector`, to wtedy spoko

Reply:
No ok